lib/testers/rule-tester.js

```diff
@@ -223,7 +251,21 @@ RuleTester.prototype = {
 
             validator.validate(config, "rule-tester");
 
-            return eslint.verify(code, config, filename);
+            // To cache AST.
+            eslint.reset();
+            eslint.on("Program", function(node) {
+                beforeAST = cloneDeeplyExcludesParent(node);
+
+                eslint.on("Program:exit", function(node) {
```

Comment:
I meant adding `api.on("onStart", ast);` and `api.on("onEnd", ast);` before and after [here](https://github.com/eslint/eslint/blob/master/lib/eslint.js#L779-L787), then the rule tester uses those.